.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+# How to contribute
+Thank you for showing interest in contributing to Pretendo Network! As an open source project, Pretendo Network welcomes contributions from developers outside of our core team. We are a large project, made up of over 100 different repositories spanning various languages and scopes. For the sake of consistency and maintainability, we have created these guidelines to outline project goals, standards, and best practices. This document will cover both code contributions as well as going over how to report any bugs or issues with a service or documentation. By adhering to these guidelines, contributors can ensure that their contributions align with the project's vision and meet the quality standards expected by the community.
+
+### Contents
+- [Prerequisites](#prerequisites)
+- [Making Issues](#making-issues)
+- [Code Changes](#code-changes)
+	- [Consistency](#consistency)
+	- [Quality and Substance](#quality-and-substance)
+- [Commits](#commits)
+	- [Scale and Scope](#scale-and-scope)
+	- [Messages](#messages)
+- [Pull Requests](#pull-requests)
+
+# Prerequisites
+Before diving into the contribution process, we recommend familiarizing yourself with our project's existing repositories. By studying our existing codebases, issues, and pull requests you will get a better sense of direction of how we operate. Additionally, please review our [Code of Conduct](CODE_OF_CONDUCT.md) to ensure a respectful and inclusive environment for all participants.
+
+Given that our project's main goal is to emulate Nintendo Network, it is also highly recommended that you familiarize yourself with how these services operate. Information on these topics, and our libraries, can be found both on our [wiki](https://nintendo-wiki.pretendo.network/docs/) and our (in progress) [developer documentation](https://developer.pretendo.network/home).
+
+> [!NOTE]
+> Developer documentation is highly work in progress. Do not expect to find everything there at this time. Please refer to the wiki first and foremost.
+
+Finally, we recommend joining our [Discord server](https://invite.gg/pretendo) to see any roadmap plans or ask specific questions from our developers.
+
+# Making Issues
+Before contributing any code, you should understand how we use issues and how to make them in a way which ensures you will get the response you want.
+
+Issues are used by all repositories for most interactions. Issues are used for reporting actual issues with the codebase, as well as making feature requests and asking general questions. When making an issue, please use one of the provided issue templates. Using the provided templates keeps things consistent, as well as makes it easier for any of our 3rd party tools to interact with issues. If a template does not exist for what you wish to do, create a feature request for one. Not using an issue template may result in your issue being closed without completion.
```

Comment:
> If a template does not exist for what you wish to do, create a feature request for one. Not using an issue template may result in your issue being closed without completion.

This feels like an unnecessary barrier to contribution, implying that they have to wait for a feature request to be implemented before even reporting their issue.

Reply:
There are issue templates for all the common use cases (even going so far as to split bug reports into bugs and regressions). The logic here is that if you need to use issues for something not covered by the existing templates then it shows we may need to add one

The point of the templates is to make it so that issues are structured the same across all issues, which should make reading them faster as we'd know where the relevant information is already

Do you have a better suggestion? (Genuine question)

Also to be clear the "it may be closed without completion" part was mostly aimed at people making issues that have templates available and just don't use them (like filing a bug report without using the template)

Reply:
Maybe ask to please *also* file a feature request? So they can still file their thing immediately.

Reply:
Take a small PR to implement a small fix - like, a typo in a print statement, do we *really* need an issue for that?

Do we *need* an issue with no comment or discussion for a PR that will definitely be merged? I feel like its unnecessary - issues IMO should be for 'these need to be implemented in the future but I'm not going to do it now'